addon/addon/index.js

```diff
@@ -1,11 +1,25 @@
 /* globals requirejs, require */
 
-import Ember from 'ember';
 import { assert, deprecate, warn } from '@ember/debug';
 import EmberObject from '@ember/object';
 import { dasherize, classify, underscore } from './string';
 import { DEBUG } from '@glimmer/env';
 import classFactory from './utils/class-factory';
+import {
+  macroCondition,
+  dependencySatisfies,
+  importSync,
+} from '@embroider/macros';
+
+let getOwner;
+
+if (macroCondition(dependencySatisfies('ember-source', '>= 4.11'))) {
```

Comment:
Since this PR is now breaking anyway (which I agree with), how about not doing these macroCondition shenanigans and just supporting ember >= 4.11?

Reply:
done! I like this plan